lib/twiglet/validator.rb

```
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+require 'json-schema'
+require 'json'
+
+module Twiglet
+  class Validator
+    def initialize(schema)
+      @schema = schema
+    end
+
+    def self.from_file(file_path)
+      new(JSON.parse(File.read(file_path)))
+    end
+
+    def validate(message)
+      return unless block_given?
+
+      yield message unless JSON::Validator.validate(@schema, message)
```

I'm not sure that this should silently fail here, the developer should get some feedback on what went wrong - the `validate!` or `fully_validate` methods are supposed to provide this feedback.